### PR TITLE
WIP: Python3 compatibility for Content Copy Tool

### PR DESCRIPTION
**Pull request by therealmarv**
wip (needs better testing first)

depending on
* https://github.com/openstax/cnx-tools/pull/8
* https://github.com/openstax/cnx-tools/pull/9
* https://github.com/openstax/cnx-tools/pull/11
* https://github.com/openstax/cnx-tools/pull/17

### CODE REVIEW

**Review by edwoodward — APPROVED** [review]

**Comment by edwoodward** [comment]
@therealmarv  I'll review it again once it is out of Draft

**Review by edwoodward — APPROVED** [review]

**Comment by therealmarv** [comment]
New corner case issue with a new "UTF-8" error. See issue #19. Unfortunately I could not reproduce this error on my machine.

**Comment by brittweinstein** [comment]
@tomjw64 please review when you can

**Comment by therealmarv** [comment]
Just want to emphasize: This PR needs testing with Python2 + Python3 before it gets merged.
I've seen corner cases in either Python2 or Python3 (forget were it was) were this PR still fails or was behaving not as good as in the other Python version.

**Review by tomjw64 — CHANGES_REQUESTED** [review]
Because some issues have been observed with this update to python3, and there is little-to-no benefit in merging this until 6 months from now, as Marvin himself has stated, this need ample testing before being merged. In addition, with this timeline, since this tool uses the legacy editor to do what it does, some other solution needs to come forth, perhaps through Neb. There is a reasonable chance that by the time it really becomes worth it to update to python3, this tool will be virtually obsolete.

**Comment by brittweinstein** [comment]
@therealmarv I'm moving this back to in progress per Thomas's comment

**Review by mmulich — APPROVED** [review]
I don't see any issues with the code in this PR. I've got tried it myself, but again it looks 👀 just fine.

**Comment by mmulich** [comment]
Should this really have been reopened? What appears to be the hold up on this?

**Comment by therealmarv** [comment]
It needs more testing and probably fixes. It's some months ago but with the new code either Python2 or Python3 was failing on edge cases. The old code (python2 only) works always. Basically the work on this code is 95% done but needs more testing and fixes on edge cases on BOTH python versions.
TL;DR: It's not finished.
